storage/txcache/txListBySenderMap.go

```diff
@@ -108,6 +108,11 @@ func (txMap *txListBySenderMap) removeSender(sender string) bool {
 		return false
 	}
 
+	// TODO / bugfix: when we concurrently try to remove the same sender,
+	// we might end up decrementing the counter twice.
+	// Possible solution: use an "onItemRemoved" callback in the "backingMap"
```

Comment:
To prevent double decreasing, might use a mutex or atomic function for that in order to make the decrementation concurrent safe.